dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/HeartbeatHandler.java

```diff
@@ -67,8 +69,15 @@ public void received(Channel channel, Object message) throws RemotingException {
         if (isHeartbeatRequest(message)) {
             Request req = (Request) message;
             if (req.isTwoWay()) {
-                Response res = new Response(req.getId(), req.getVersion());
-                res.setEvent(HEARTBEAT_EVENT);
+                Response res;
+                if (req instanceof HeartBeatRequest) {
+                    res = new HeartBeatResponse(req.getId(), req.getVersion());
+                    res.setEvent(HEARTBEAT_EVENT);
+                    ((HeartBeatResponse) res).setProto(((HeartBeatRequest) req).getProto());
+                } else {
+                    res = new Response(req.getId(), req.getVersion());
+                    res.setEvent(HEARTBEAT_EVENT);
```

Comment:
res.setEvent(HEARTBEAT_EVENT);    
this line can put it outside.
Then under what circumstances will enter this else?

Reply:
> Then under what circumstances will enter this else?

I also have the same question.